recommenders/models/sar/sar_singlenode.py

```diff
@@ -446,6 +490,80 @@ def get_item_based_topk(self, items, top_k=10, sort_top_k=True):
         # drop invalid items
         return df.replace(-np.inf, np.nan).dropna()
 
+    def get_user_based_topk(self, users, top_k=10, sort_top_k=True):
+        """Get top K similar users to provided seed users based on similarity metric defined.
+        This method will take a set of users and use them to recommend the most similar users to that set
+        based on the similarity matrix fit during training.
+
+        Args:
+            users (pandas.DataFrame): DataFrame with user, item (optional), and rating (optional) columns
+            top_k (int): number of top users to recommend
+            sort_top_k (bool): flag to sort top k results
+
+        Returns:
+            pandas.DataFrame: similar users to the users provided in users
+        """
```

Comment:
One question @danb27, the computation above of the user coocurrence is only used when we want to use the method `get_user_based_topk`?

Reply:
correct. in `fit()`, if we set `compute_user_similarity=True`, `user_cooccurrence` will be calculated and used to create `self.user_similarity`, which is required to run `get_user_based_topk`. 

I went back and forth on whether to compute user similarity always, but decided to create the `compute_user_similarity` parameter so that the user could keep performance the same as before the PR if they do not require the added functionality. 

Happy to have my mind changed if you all would prefer to have this computation occur every time so that you can always call any of the methods, regardless of how you fit.

Reply:
what would be the change required and the performance difference if we do all the computation of user_coocurrence inside the function `get_user_based_topk`?

Reply:
> I am envisioning the use case (which is my current use case) where someone wants to store similar users for every user in their dataset, so it is necessary to call this method n_users times.

Ideally, it would be great that given a list of user_ids (or DF), the system returns a DF (or list of list, whatever) but the top k most similar users for each input user_id in the list. It looks that is not happening, right?

Some questions:
1. How can we create this result?
2. Right now we can input a list of user_ids, what is the meaning of the output?

Reply:
1)

Currently, I call `get_item_based_topk()` like this: 

```
for item_id in item_ids:
    temp_df = pd.DataFrame({'item_id': [item]}, dtype=DTYPE)
    similar_items_df = model.get_item_based_topk(temp_df, k=K)
    ...
```
With my PR, I would be able to do something very similar for `get_user_based_topk()` 

We could implement this as a different method, because what currently occurs if you give this method a list of user ids is the following:

2. From the docstring of `get_item_based_topk()`: " This method will take a set of items and use them to recommend the most similar items to that set based on the similarity matrix fit during training. This allows recommendations for cold-users (unseen during training), note - the model is not updated." The same is true for the new `get_user_based_topk()` method, it returns the most similar users given a set of other users.

Reply:
I think it would be valuable to have both item and user similar. Given a list of items, I would like to get a list of the most similar ones for each item, and given a list of users, I would like to get a list of the most similar users for each user.